app/views/products/_product.html.erb

```diff
@@ -1,9 +1,9 @@
 <div id="<%= dom_id product %>">
 
 <div class="slider">
-  <%= image_tag(product.images.to_a.first, class: 'slider-img') if product.images.to_a.first %>
-  <%= image_tag(product.images.to_a.second, class: 'slider-img') if product.images.to_a.second %>
-  <%= image_tag(product.images.to_a.third, class: 'slider-img') if product.images.to_a.third %>
+<% product.images.to_a.each do |img| %>
+  <%= image_tag(img, class: 'slider-img') if img.present? %>
+  <% end %>
```

Comment:
is `to_a` required? 
is `img.present?` check required?

Reply:
to_a and img.present? is not required